packages/keyring-eth-hd/src/index.ts

```diff
@@ -0,0 +1,590 @@
+   * @returns The Uint8Array mnemonic.
+   */
+  #stringToUint8Array(mnemonic: string): Uint8Array {
+    const indices = mnemonic.split(' ').map((word) => wordlist.indexOf(word));
+    return new Uint8Array(new Uint16Array(indices).buffer);
+  }
+
+  /**
+   * Convert a mnemonic to a Uint8Array mnemonic.
+   *
+   * @param mnemonic - The mnemonic seed phrase.
+   * @returns The Uint8Array mnemonic.
+   */
+  #mnemonicToUint8Array(
+    mnemonic: Buffer | JsCastedBuffer | string | Uint8Array | number[],
+  ): Uint8Array {
+    let mnemonicData: unknown = mnemonic;
+    // when encrypted/decrypted, buffers get cast into js object with a property type set to buffer
+    if (isJsCastedBuffer(mnemonic)) {
+      mnemonicData = mnemonic.data;
+    }
+
+    if (
+      // this block is for backwards compatibility with vaults that were previously stored as buffers, number arrays or plain text strings
+      typeof mnemonicData === 'string' ||
+      Buffer.isBuffer(mnemonicData) ||
+      Array.isArray(mnemonicData)
+    ) {
+      let mnemonicAsString: string;
+      if (Array.isArray(mnemonicData)) {
+        mnemonicAsString = Buffer.from(mnemonicData).toString();
+      } else if (Buffer.isBuffer(mnemonicData)) {
+        mnemonicAsString = mnemonicData.toString();
+      } else {
+        mnemonicAsString = mnemonicData;
+      }
+      return this.#stringToUint8Array(mnemonicAsString);
+    } else if (
+      mnemonicData instanceof Object &&
+      !(mnemonicData instanceof Uint8Array)
+    ) {
+      // when encrypted/decrypted the Uint8Array becomes a js object we need to cast back to a Uint8Array
+      return Uint8Array.from(Object.values(mnemonicData));
+    }
+
+    assert(mnemonicData instanceof Uint8Array, 'Expected Uint8Array mnemonic');
+    return mnemonicData;
+  }
+
+  /**
+   * Get the private key for the specified account.
+   *
+   * @param address - The address of the account.
+   * @param opts - The options for selecting the account.
+   * @returns The private key of the account.
+   */
+  #getPrivateKeyFor(
+    address: string,
+    opts?: HDKeyringAccountSelectionOptions,
+  ): Uint8Array | Buffer {
+    if (!address) {
+      throw new Error('Must specify address.');
+    }
+    const wallet = this.#getWalletForAccount(address, opts);
+    assert(wallet.privateKey, 'Missing private key');
+    return wallet.privateKey;
+  }
+
+  /**
+   * Get the wallet for the specified account.
+   *
+   * @param account - The address of the account.
+   * @returns The wallet for the account as HDKey.
+   */
+  #getWalletForAccount(account: string): HDKey;
+
+  /**
+   * Get the wallet for the specified account and app origin.
+   *
+   * @param account - The address of the account.
+   * @param opts - The options for selecting the account.
+   * @returns A key pair representing the wallet.
+   */
+  #getWalletForAccount(
+    accounts: string,
+    opts: { withAppKeyOrigin: string },
+  ): { privateKey: Buffer; publicKey: Buffer };
```

Comment:
This internal method is overloaded because the objects returned can vary significantly based on whether an app key origin is specified (i.e. a Buffer keypair is returned instead of an entire HDKey instance).

The actual signature of the implementation is left unchanged.